src/api/hooks/useGetDelegatorStakeInfo.ts

```diff
@@ -0,0 +1,23 @@
+import {Types} from "aptos";
+import {useState, useEffect} from "react";
+import {getStake} from "..";
+import {useGlobalState} from "../../GlobalState";
+
+export function useGetDelegatorStakeInfo(delegatorAddress: Types.Address) {
+  const [state, _] = useGlobalState();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [stake, setStake] = useState<Types.MoveValue[]>([]);
+
+  useEffect(() => {
+    if (stake !== undefined) {
+      const fetchData = async () => {
+        const stakes = await getStake(state.network_value, delegatorAddress);
+        setStake(stakes);
```

Comment:
a bit confused here, looks like when `stake` is defined, we get the stake? and then set the stake state, which triggers the useEffect again and start this whole cycle?